src/app/clusters/ota-requestor/OTARequestor.cpp

```diff
@@ -133,13 +136,25 @@ void OTARequestor::OnQueryImageResponse(void * context, const QueryImageResponse
             return;
         }
 
-        MutableByteSpan updateToken(requestorCore->mUpdateTokenBuffer);
-        CopySpanToMutableSpan(update.updateToken, updateToken);
-        requestorCore->mUpdateVersion = update.softwareVersion;
-        requestorCore->mUpdateToken   = updateToken;
+        if (update.softwareVersion > currentSoftwareVersion)
+        {
+            ChipLogDetail(SoftwareUpdate, "Updating to newer version");
```

Comment:
If you're going to log this, you might as well log the old and new version numbers. Would be helpful for debugging

"Update available from __ to __"

Reply:
Done